Doc/library/typing.rst

```diff
@@ -1236,47 +1236,51 @@ These are not used in annotations. They are building blocks for creating generic
 
     :class:`Type variable <TypeVar>` tuple.
 
-    Usage::
+    A vanilla type variable enables parameterization with a single type. A type
+    variable tuple, in contrast, allows parameterization with an
+    *arbitrary* number of types by acting like an *arbitrary* number of type
+    variables wrapped in a tuple. For example::
 
-        Ts = TypeVarTuple('Ts')
         Shape = TypeVarTuple('Shape')
 
-    Vanilla type variables enable parameterization with a *fixed* number of
-    types. Type variable tuples, on the other hand, allow parameterization with
-    an *arbitrary* number of types by acting like an *arbitrary* number of type
-    variables wrapped in a tuple. For example::
+        class Array(Generic[*Shape]): ...
 
-        class Array(Generic[*Shape]):
-            pass
+        array_1d: Array[int] = Array()       # 1 type parameter: fine
+        array_2d: Array[int, int] = Array()  # 2 type parameters: also fine
+        array_0d: Array[()] = Array()        # 0 type parameters: also fine
 
-        from typing import NewType
-        Height = NewType('Height', int)
-        Width = NewType('Width', int)
+    Note the use of the unpacking operator ``*`` in ``Generic[*Shape]``.
+    Conceptually, you can think of ``Shape`` as a tuple of type variables
+    ``(T1, T2, ...)``. ``Generic[*Shape]`` would then become
+    ``Generic[*(T1, T2, ...)]``. The ``*`` brings the type variables out of the
```

Comment:
Maybe add "which is equivalent to `Generic[T1, T2, ...]`". Because that's what we're really after.

Reply:
I was a bit worried that `Generic[T1, T2, ...]` might make some users think that this was valid syntax itself - and I didn't want to break the flow of reading by adding a parenthetical that it's not. But if you think it's fine, I'm happy to go for it.